trace-dispatcher/doc/trace-dispatcher.md

```diff
@@ -0,0 +1,798 @@
+
+The __tracer namespace__ appears in the following contexts:
+
+* __documentation__, where it defines the overall structure of the generated documentation output,
+* __configuration__, where it allows referring to tracers we want to reconfigure in some way, such as changing their severity,
+* __trace-outs__, where the __metrics__ and __messages__ carry the __tracer name__.
+
+Given a __tracer__ with a particular __tracer name__, we can derive a tracer with an extended __tracer name__:
+
+```haskell
+appendName :: Monad m => Text -> Trace m a -> Trace m a
+```
+
+As an example, consider the following tracer:
+
+```
+appendName "specific" $ appendName "middle" $ appendName "general" tracer
+```
+
+..which will have the name `["general", "middle", "specific"]`.
+
+## Trace context
+
+As mentioned in the overview, __traces__ are interpreted by the __dispatcher__ in a __trace context__.  This context consists of two parts:
+
+* the __logging context__ of the trace, as introduced by __trace combinators__, and
+* the __configuration context__ coming from the program configuration (initial or runtime).
+
+Both pieces meet together, to inform the following decisions:
+
+1. __trace filtering__ -- whether a trace reaches particular __trace-outs__ or not,
+2. __trace presentation__ -- which detail level is used during transformation of the __trace__ into __messages__.
+
+The __logging context__ of the trace is defined as follows:
+
+1. __trace filtering__ -- by __privacy__, __severity__ and __namespace__ context,
+2. __trace presentation__ -- by __detail level__ context.
+
+Severity an detail level can be configured.
+
+## Filter context
+### Severity
+
+__Severity__ is expressed in terms of the enumeration provided by [section 6.2.1 of RFC 5424](https://tools.ietf.org/html/rfc5424#section-6.2.1):
+
+```haskell
+data SeverityS
+    = Debug | Info | Notice | Warning | Error | Critical | Alert | Emergency
+```
+
+..which ranges from minimum (`Debug`) to the maximum (`Emergency`) severity, and allows ignoring messages with severity level _below_ a configured global __severity cutoff__.
+
+The following __trace combinators__ affect __annotated severity__ of a trace:
+
+```haskell
+withSeverity :: Monad m => (a -> Severity) -> Trace m a -> Trace m a
+setSeverity  :: Monad m => Severity        -> Trace m a -> Trace m a
+
+-- Unconditional annotation:
+tracer   = setSeverity Notice trAddBlock
+
+-- Conditional annotation:
+tracer'' = withSeverity (\case
+                           IgnoreBlockOlderThanK{}          -> Warning
+                           IgnoreBlockAlreadyInVolatileDB{} -> Notice)
+                        trAddBlock
+```
+
+If the combinators are applied multiple times to a single trace, only the outermost application affects it -- the rest of them is ignored.
```

Comment:
I wonder if in practice you even want to only partially override the severity i.e. keep the severity unchanged:

```haskell
tracer'' = withSeverity (\case
                           IgnoreBlockOlderThanK{}          -> Warning
                           IgnoreBlockAlreadyInVolatileDB{} -> Notice
                           InternalError                    -> error "I want to leave the severity unchanged")
                        trAddBlock
```

Reply:
In new binding to the node there has been no need for a partial binding. If needed you can use setSeverity for individual messages.